Exercise3Additional.sql

```diff
@@ -0,0 +1,40 @@
+/*EXERCISE3
+Additional 1:
+Checklists briefly described in Exercise 1 are being created for each Plot.
+So create function which checks if Plot has created Checklist and returns �Yes� � if so and �No� � otherwise.
+Function should have Plot Id as input parameter. To see if there is a Checklist for it, verify whether related ApplicationData. 
+CheckListHeader record exists which has one-to-one relation with ApplicationData.HBPlot. 
+Call this function along with others in GetPlotInfo and show result as new output parameter.
+
+Additional 2:
+As it briefly described in Exercise 2, various issues may appear during Plot building process.
+Plots may have many Issues, single Issue could be linked to many Plots - so there is many-to-many relation between ApplicationData.
+Issue and ApplicationData.HBPlot tables.  In database it is resolved through cross-reference table ApplicationData.IssueXRefPlot. 
+
+Please, create JS function which counts number of Opened Issues for Plot (as before, Plot Id is input parameter).
+Basically, to count Plot�s Issues it is enough to count related XRef records, but task is to count Issues with Status = �Open�.
+So SQL query has to JOIN Issue table additionally (and ApplicationData.IssueStatus optionally). 
+Call this function from GetPlotInfo and show result as new output parameter.
+*/
+
+/*EXERCISE3 ADDITIONAL1*/
+DECLARE @yourPlotID int = 30;
+IF EXISTS (SELECT [ApplicationData].[CheckListHeader].[ForeignTableId]
```

Comment:
this approach is good to practice with SQL and stored procedures. 
but if there will be a need to use this query later in our JS Editor, EXISTS can be included directly in SELECT:
SELECT IIF(EXISTS(SELECT...),'YES','NO') AS RecordExist
FROM...